File: src/ProjectR.Scripting/content/scripts/spells/lua/LightAndShadowIntertwined.lua
function GetTargetType()
    return Enemies
end

function GetName()
  return "Light and Shadow Intertwined"
end

function GetDescription()
  return "Twilight surrounds your enemies.\nReduces SPD and can inflict PAR."
end

function IsSupportSpell()
    return false
end

function GetMPCost()
    return .64
end

function GetDelay()
  return .4
end

function GetSpellType()
  return Magical
end

function GetMasteries()
  return { DRK, HOL }
end

function SpellEffect()
  local damage = (3.75 * aMD * ((aHOL + aDRK) / 200) - 1.5 * dMR) * (200 / (dHOL + dDRK))
  d:BuffStat(SPD, -.15)
  d:ApplyDebuff(PAR, 35)
  d:TakeDamage(damage)
end

File: src/ProjectR.Scripting/content/scripts/spells/LightAndShadowIntertwined.cs
using System.Collections.Generic;
using ProjectR.Interfaces;
using ProjectR.Interfaces.Model;
using ProjectR.Interfaces.Model.Stats;

namespace ProjectR.Scripting.Spells
{
    public class LightAndShadowIntertwined : SpellScriptBase
    {
        public override string Name { get { return "Light and Shadow Intertwined"; } }
        public override string Description { get { return "Twilight surrounds your enemies.\nReduces SPD and can inflict PAR."; } }

        public override TargetType TargetType { get { return TargetType.Enemies; } }
        public override IList<EleMastery> Masteries { get { return new EleMastery[] { EleMastery.DRK, EleMastery.HOL, }; } }
        public override SpellType SpellType { get { return SpellType.Magical; } }
        public override bool IsSupportSpell { get { return false; } }

        public override double MPCost { get { return 64; } }
        public override double Delay { get { return .4; } }

        public override void SpellEffect(ICharacter caster, ICharacter target)
        {
            var damage = (3.75 * cMD * ((cHOL + cDRK) / 150) - 1.5 * tMR) * (200 / (tHOL + tDRK));
            BuffStat(Stat.SPD, -.15);
            TryToApplyDebuff(DebuffResistance.PAR, 35);
            DealDamage(damage);
        }
    }
}